src/app/shared/services/snapshot.service.ts

```diff
@@ -38,6 +38,10 @@ export class SnapshotService extends BaseBackendCachedService<Snapshot> {
       });
   }
 
+  public markForRemoval(snapshot: Snapshot): Observable<Snapshot> {
+    return this.snapshotTagService.markForRemoval(snapshot) as Observable<Snapshot>;
```

Comment:
why casting? I think this method return something like success: true. Can you check that?